2.6/README.md

```diff
@@ -23,26 +23,111 @@ modules for their web applications. There is no guarantee for any specific npm o
-    ```
+Source-to-Image framework and scripts
+-------------------------------------
+This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/4.4/builds/build-strategies.html#images-create-s2i_build-strategies)
+(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
+which makes it easy to write images that take application source code as
+an input, use a builder image like this Ruby container image, and produce
+a new image that runs the assembled application as an output.
 
-**Accessing the application:**
+To support the Source-to-Image framework, important scripts are included in the builder image:
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application and places them into appropriate directories inside the image. It utilizes some common patterns in Ruby application development (see the **Environment variables** section below).
+* The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts). It configures `puma` server and runs a web server on port 8080 using `rackup`.
+
+Building an application using a Dockerfile
+---------------------
+Compared to the Source-to-Image strategy, using a Dockerfile is a more
+flexible way to build a Ruby container image with an application.
+Use a Dockerfile when Source-to-Image is not sufficiently flexible for you or
+when you build the image outside of the OpenShift environment.
+
+To use the Ruby image in a Dockerfile, follow these steps:
+
+#### 1. Pull a base builder image to build on
+
+```
+podman pull ubi8/ruby-26
+```
+
+An RHEL7 image `ubi8/ruby-26` is used in this example.
+
+#### 2. Pull and application code
+
+ An example application available at https://github.com/sclorg/rails-ex.git is used here. Feel free to clone the repository for further experiments.
+
+```
+git clone https://github.com/sclorg/rails-ex.git app-src
 ```
-$ curl 127.0.0.1:8080
+
+#### 3. Prepare an application inside a container
+
+This step usually consists of at least these parts:
+
+* putting the application source into the container
+* installing the dependencies
+* setting the default command in the resulting image
+
+For all these three parts, users can either setup all manually and use commands `ruby`, `bundle` and `rackup` explicitly in the Dockerfile ([3.1.](#31-to-use-your-own-setup-create-a-dockerfile-with-this-content)), or users can use the Source-to-Image scripts inside the image ([3.2.](#32-to-use-the-source-to-image-scripts-and-build-an-image-using-a-dockerfile-create-a-dockerfile-with-this-content); see more about these scripts in the section "Source-to-Image framework and scripts" above), that already know how to set-up and run some common Perl applications.
```

Comment:
```
that already know how to set-up and run some common Perl applications
```
`know` cannot be used with inanimate object. Or maybe the sentence is too confusing? Why Perl is mentioned?

Reply:
Right, the english wording is incorrect, but let's adress that in some other PR.

Reply:
it's a common part of README's across containers (see for example node.js and others), I don't see there any problem, scripts know how to set-up .....not users

Reply:
Well, like I wrote previously: `know` cannot be used with inanimate object.
Also, I'm not saying you don't relay the message, but that it's not quite clear, at least for me, to understand exactly the message is (as it's gramatically incorrect).

Reply:
ok thanks, do you have some suggestions how to improve it?